CalibTracker/SiStripCommon/interface/Book.h

```diff
@@ -62,7 +62,7 @@ class Book {
   void fill( double_t X, const poly<std::string>& names, 
 	     uint_t NbinsX, double_t Xlow, double_t Xup, double_t W=1 ) 
   { 
-    BOOST_FOREACH(string_t name, std::make_pair(names.begin(),names.end())) {
+    for(auto const& name : names) {
```

Comment:
@guitargeek are you sure that those loops are equivalent? `std::make_pair` will return a pair of first and last elements while the new version will iterate over every value in `names`.

Reply:
You can already see from the types that it can't be as you thought. The `std::make_pair` gives you a pair of iterators, but the elements are strings. Therefore I understood that this `BOOST_FOREACH` can also take a pair of iterators, where the first one is the "begin" and the second is the "end".

Reply:
Here you have an example:
```
#include <boost/foreach.hpp>
#include <iostream>
#include <vector>

int main ()
{
    std::vector<int> v {0,1,2,3,4,5,6,7,8,9};

    for (int x : v) {
        std::cout << x << " ";
    }
    std::cout << std::endl;

    BOOST_FOREACH(int x, std::make_pair(v.begin(), v.end())) {
        std::cout << x << " ";
    }
    std::cout << std::endl;
}
```
This program returns as I expected:
```
0 1 2 3 4 5 6 7 8 9 
0 1 2 3 4 5 6 7 8 9 
```

Reply:
Ok, cool. I didn't realise that `BOOST_FOREACH` will loop between two iterators.